ipaserver/plugins/otptoken.py

```diff
@@ -311,13 +311,12 @@ def pre_callback(self, ldap, dn, entry_attrs, attrs_list, *keys, **options):
         # If owner was not specified, default to the person adding this token.
         # If managedby was not specified, attempt a sensible default.
         if 'ipatokenowner' not in entry_attrs or 'managedby' not in entry_attrs:
-            result = self.api.Command.user_find(
-                whoami=True, no_members=False)['result']
-            if result:
-                cur_uid = result[0]['uid'][0]
+            cur_dn = DN(self.api.Backend.ldap2.conn.whoami_s()[4:])
```

Comment:
I would just add a comment explaining the [4:] (the result is a string starting with dn: uid=... and as we want to remove the "dn: " at the beginning, we need to remove the 4 first chars).

Reply:
@flo-renaud we have this pattern in other places:
```
$ git grep conn.whoami_s
ipapython/ipaldap.py:        bind_dn = self.conn.whoami_s()[4:]
ipaserver/plugins/otptoken.py:            cur_dn = DN(self.api.Backend.ldap2.conn.whoami_s()[4:])
ipaserver/plugins/whoami.py:        dn = DN(ldap.conn.whoami_s()[4:])
```
I'm not sure it makes sense to add this info everywhere. It is documented in LDAP RFCs: https://tools.ietf.org/html/rfc4513 section 5.2.1.8. tells that authzid is
```
      authzId = dnAuthzId / uAuthzId

      ; distinguished-name-based authz id
      dnAuthzId =  "dn:" distinguishedName

      ; unspecified authorization id, UTF-8 encoded
      uAuthzId = "u:" userid
      userid = *UTF8 ; syntax unspecified
```

and "WHO AM I?" LDAP RFC https://tools.ietf.org/html/rfc4532 explicitly refers to RFC4513 for the format of the returned identity.

Reply:
Hi @abbra 
my point is that a short comment can sometimes save others time. We don't all have the same LDAP knowledge and experience with LDAP RFCs.
I don't ask to change the code everywhere but when touching a portion of code, it seems good practice to me to try to enhance the clarity. My 2 cents...